docs/DeveloperGuide.md

```diff
@@ -1,34 +1,853 @@
+
+1. The user enters `help`, and the program displays the help message. 
+Since this command does not modify the data, 
+the external file remains unchanged.
+
+1. The user enters `addmod CS1010 10`, 
+and the program adds the new module correspondingly. 
+Since this is a valid command which modifies the data,
+ the `Storage` object appends `addmod CS1010 10` to the external file.
+
+1. The user enters `exit`, and the program terminates.
+
+##### Second Use
+The same user starts the application again (on the same device as the first use).
+
+1. `ModTracker` creates a new `Storage` object 
+with the same specified file path at `data/modtracker.txt`.
+   
+1. The `Storage` object checks that the file at `data/modtracker.txt` exists,
+and reads the first line of the file to obtain the user's name.
+
+1. The `Storage` object passes the username to `ModTracker`, 
+which calls `Ui` to greet the user.
+
+1. `ModTracker` then obtains the remaining lines in the file via `Storage`,
+and passes it to `Parser` to load the data into `ModuleList` and `TaskList`.
+
+1. The program prompts the user for further inputs,
+and it continues as per normal, with the data loaded.
+
+The following sequence diagram shows how the storage feature works,
+with a focus on how it obtains the username:
+
+![Storage Sequence Diagram](diagrams/Storage%20Sequence%20Diagram.svg)
+
+#### Design Considerations
+
+* **Alternative 1 (current choice)**: Saves the user input to the file
+    * Pros: Easy to implement as code from `Parser` can be reused
+    * Cons: Increases coupling as it depends on the `Parser` class 
+    to make sense of the data
+* **Alternative 2**: Parses input into a different format 
+storing the different modules, time expected, time spent and tasks
+    * Pros: Independent of how `Parser` takes in user input
+    * Cons: Requires additional work to parse data into the required
+    storage format
+
+
+### 4.2 Add Module <a name="4.2"></a>
+
+#### Current Implementation
+
+The add module feature will implement the checks of the module code input by the user through the following operations:
+
+`checkIfModuleValid()` – Checks if the module code starts with 2 - 3 characters, followed by 4 digits, followed by an 
+                         optional character.
+
+`checkIfModuleExist()` – Checks if the module already exists in the list of modules
+
+If the module code is valid and does not exist, the add module feature will proceed to create the module and adds it to 
+the list of modules.  
+
+> **_NOTE:_**  A module is identfied by its module code
+
+The following activity diagram summarizes what happens when a user executes `addmod CS2113T` command for the first time.  
+
+![addmod](diagrams/addmod%20activity%20diagram.png)
+
+Given below is a detailed explanation on how the add module mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time and hence, the list of modules is empty. 
```

Comment:
Should the numbering schemes be more coherent? It seems like there are some parts with just numbers and others with step + number
![image](https://user-images.githubusercontent.com/61812576/97655339-d0321f00-1a9f-11eb-9c5d-e8b9b07fb2f8.png)